README.md

```diff
@@ -1,2 +1,98 @@
 # patch-filters
 special project to host GSOC project Patch Suppressions 
+
+## PatchFilter Report Setup
+
+### Requirements
+
+- Checkstyle repository which including patch-filter need to be cloned.
+- [Contribution repository](https://github.com/checkstyle/contribution) need to be cloned.
+
+### Clone
+
+Just clone this repository to your local.
+
+```bash
+$ git clone git@github.com:checkstyle/patch-filters.git
+```
+
+Or use HTTPS mode.
+```bash
+$ git clone https://github.com/checkstyle/patch-filters.git
+```
+### local install
+
+Because patch-filter has not been merged by Checkstyle, so you should local install this repository——patch-filters
+
+```bash
+$ cd /path/to/patch-filters
+$ mvn clean install
+```
+
+then add this dependency to checkstyle
+
+```xml
+<dependency>
+  <groupId>com.github.checkstyle</groupId>
+  <artifactId>patch-filter</artifactId>
+  <version>1.0-SNAPSHOT</version>
+</dependency>
+```
+then create a new branch, for example `patch-filter`
+
+```bash
+$ cd /path/to/checkstyle
+$ git checkout -b patch-filter
+$ git add .
+$ git commit -m "add patch-filter dependency"
+```
+
+
+### Open this repo in IDEA or other IDE
+
+You can find `GeneratePatchFileTest` in `/path/to/patch-filters/src/test/java/com/github/checkstyle/generatepatchfile/`,
+then add follow code in this Class:
+```java
+package com.github.checkstyle.generatepatchfile;
+
+import org.junit.Test;
+
+public class GeneratePatchFileTest {
+    @Test
+    public void generatePatchFileTest() {
```

Comment:
Remove annotatio and convert it to `main` method. Put such class to codebase to use it out of the box

Reply:
ok, It means to make it as an executable class, right?

Reply:
done